src/lib/storage/table.cpp

```diff
@@ -85,4 +86,32 @@ std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) { return _chunks.at(ch
 
 std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const { return _chunks.at(chunk_id); }
 
+void Table::compress_chunk(const ChunkID chunk_id) {
+  const auto input_chunk = get_chunk(chunk_id);
+  const auto column_count = input_chunk->column_count();
+  std::vector<std::thread> threads;
+  threads.reserve(column_count);
+  const auto compressed_chunk = std::make_shared<Chunk>();
+  std::vector<std::shared_ptr<AbstractSegment>> compressed_segments(column_count);
+
+  for (ColumnID index{0}; index < column_count; ++index) {
+    const auto segment = input_chunk->get_segment(index);
+    threads.emplace_back([&, index, segment] {
```

Comment:
For the sake of consistency I would suggest to spell out all variables that you want to bring into the lambda scope or to  use just the `&`. 

```suggestion
    threads.emplace_back([index, segment, compressed_segments] {
```